### PR TITLE
synapseclient/reticulate documentation

### PR DESCRIPTION
**Pull request by jkiang13**
Examples in vignette style for using the Python synapseclient with R through reticulate.

As rendered:
[synapseclient_reticulate.pdf](https://github.com/Sage-Bionetworks/synapsePythonClient/files/4806205/synapseclient_reticulate.pdf)

### CODE REVIEW

**Comment by thomasyu888** [comment]
This would be an awesome thing to review at the code_review meeting

**Comment by jkiang13** [comment]
virutalenv conda instructions

**Comment by jkiang13** [comment]
Table asDataFrame issues

**Comment by jkiang13** [comment]
other use cases dependencies, e.g. shiny apps with multiple users

**Comment by jkiang13** [comment]
Using it in the context of an R package that is separately packaged, ensure things are loaded properly.

**Comment by brucehoff** [comment]
I suggest adding a section on how to use with Shiny.  In particular, it's important that the Synapse client has authenticated as the user making a request to the Shiny app'. (If the app' author fails to do this then users can end up accessing each other's Synapse content.)  An example by @karawoo :
https://github.com/Sage-Bionetworks/dccvalidator/blob/aade4ace5a3f7e11ff289c74444f32c8745b5306/R/app-server.R#L34

File: docs/reticulate.rst [hunk 1]
```diff
@@ -0,0 +1,517 @@
...
+
+Annotating Synapse Entities
+===========================
+
+This illustrates adding annotations to a Synapse entity.
+
+  .. code-block::
+
+    # first retrieve the existing annotations object
+    annotations = syn$get_annotations(project)
```
Comment (brucehoff) [inline docs/reticulate.rst:209]: In R it's customary to use "<-" rather than "=" for assignment.

File: docs/reticulate.rst [hunk 2]
```diff
@@ -0,0 +1,517 @@
...
+
+  .. code-block::
+
+    filePath <- tempfile()
+    connection <- file(filePath)
+    writeChar("some test", connection, eos = NULL)
+    close(connection)
+
+    file = synapseclient$File(filePath, name="provenance_file.txt", parent=project)
+    file <- syn$store(, activity = act)
```
Comment (brucehoff) [inline docs/reticulate.rst:241]: Is this supposed to be `file <- syn$store(file, activity = act)` ?

File: docs/reticulate.rst [hunk 3]
```diff
@@ -0,0 +1,517 @@
...
+
+    # build a Synapse table from the data frame.
+    # a schema is automatically generated
+    # note that reticulate will automatically convert from an R data frame to Pandas
+    table <- synapseclient$build_table("My Favorite Genes", project, genes)
+
+    table <- syn$store(table)
+
+Alternately the schema can be specified. At this time when using date values it is necessary
+to use string or millisecond values and explicitly specify the type in the schema due to how
```
Comment (brucehoff) [inline docs/reticulate.rst:275]: by "string or millisecond values" did you mean "millisecond values either as string or integer"?

File: docs/reticulate.rst [hunk 4]
```diff
@@ -0,0 +1,517 @@
...
+    submissions <- syn$getSubmissionBundles(eval)
+
+    # submissions are returned as a generator
+    list(iterate(submissions))
+
+Retrieving submission by id:
+
+  .. code-block::
+
+    # Not evaluating this section because of SYNPY-235
```
Comment (brucehoff) [inline docs/reticulate.rst:393]: https://sagebionetworks.jira.com/browse/SYNPY-235 is closed.  Can we remove this line?

File: docs/reticulate.rst [hunk 5]
```diff
@@ -0,0 +1,517 @@
...
+  .. code-block::
+
+    submissionStatus <- syn$getSubmissionStatus(submission)
+    submissionStatus
+
+Query an evaluation:
+
+  .. code-block::
+
+    queryString <- sprintf("query=select * from evaluation_%s LIMIT %s OFFSET %s'", eval$id, 10, 0)
```
Comment (brucehoff) [inline docs/reticulate.rst:407]: We should update this, as well as the Python doc's, to use the new Evaluation View queries.

**Comment by brucehoff** [comment]
Good work.  It will be interesting to see if this only complements or completely replaces synapser as the way to access Synapse from R.

**Pull request merged**